packages/contracts/mainnetDeployment/balancerPoolDeployment.js

```diff
@@ -0,0 +1,154 @@
+    IVault.abi,
+    deployerWallet
+  );
+
+  const priceFeed = await ethers.getContractAt('PriceFeed', PRICE_FEED)
+
+  // ZERO_ADDRESS owner means fixed swap fees
+  // DELEGATE_OWNER grants permission to governance for dynamic fee management
+  // Any other address lets that address directly set the fees
+  const oracleEnabled = true;
+  const tx1 = await factory.create(
+    NAME, SYMBOL, tokens, weights,
+    swapFeePercentage, oracleEnabled,
+    //DELEGATE_OWNER
+    impersonateAddress
+  );
+  const receipt1 = await tx1.wait();
+
+  // We need to get the new pool address out of the PoolCreated event
+  // (Or just grab it from Etherscan)
+  const events = receipt1.events.filter((e) => e.event === 'PoolCreated');
+  const poolAddress = events[0].args.pool;
+
+  // We're going to need the PoolId later, so ask the contract for it
+  const pool = new ethers.Contract(
+    poolAddress,
+    WeightedPool2Tokens.abi,
+    deployerWallet
+  );
+  const poolId = await pool.getPoolId();
+
+  // Tokens must be in the same order
+  // Values must be decimal-normalized!
+  const eth_price = await priceFeed.lastGoodPrice();
+  const weth_balance = INITIAL_FUNDING.mul(weights[1]).div(eth_price);
+  const lusd_balance = INITIAL_FUNDING.mul(weights[0]).div(toBigNum(dec(1, 18)));
+  const initialBalances = [lusd_balance, weth_balance];
+  th.logBN('Initial LUSD', lusd_balance.toString())
+  th.logBN('Initial WETH', weth_balance.toString())
+
+  const JOIN_KIND_INIT = 0;
+
+  // Construct magic userData
+  const initUserData =
+        ethers.utils.defaultAbiCoder.encode(['uint256', 'uint256[]'],
+                                            [JOIN_KIND_INIT, initialBalances]);
+  const joinPoolRequest = {
+    assets: tokens,
+    maxAmountsIn: initialBalances,
+    userData: initUserData,
+    fromInternalBalance: false
+  }
+
+  // Caller is "you". joinPool takes a sender (source of initialBalances)
+  // And a receiver (where BPT are sent). Normally, both are the caller.
+  // If you have a User Balance of any of these tokens, you can set
+  // fromInternalBalance to true, and fund a pool with no token transfers
+  // (well, except for the BPT out)
+
+  // Need to approve the Vault to transfer the tokens!
+  const weth = new ethers.Contract(
+    WETH,
+    WETH_ABI.abi,
+    deployerWallet
+  )
+  th.logBN('weth balance: ', await weth.balanceOf(deployerWalletAddress))
+  const currentWethBalance = await weth.balanceOf(deployerWalletAddress)
+  if (currentWethBalance.lt(weth_balance)) {
+    const txDepositWeth = await weth.deposit({ value: weth_balance.sub(currentWethBalance) });
+    await txDepositWeth.wait()
+  }
+  th.logBN('weth balance: ', await weth.balanceOf(deployerWalletAddress))
+  const txApproveWeth = await weth.approve(VAULT, weth_balance);
+  await txApproveWeth.wait()
+  const lusd = new ethers.Contract(
+    LUSD,
+    ERC20.abi,
+    deployerWallet
+  )
+  const txApproveLusd = await lusd.approve(VAULT, toBigNum(dec(1, 36)));
```

Comment:
Any reason for `1e136` here, or is the intent just to do a large approval that will definitely cover it?

Reply:
Good catch! It should be `lusd_balance`. I changed it at some point because the script was failing, and forgot to put it back. I’ll fix it, thanks!